src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/ActivityTracker.cs

```diff
@@ -133,18 +134,19 @@ public void OnStart(string providerName, string activityName, int task, ref Guid
         ///
         /// If activity tracing is not on, then activityId and relatedActivityId are not set
         /// </summary>
-        public void OnStop(string providerName, string activityName, int task, ref Guid activityId)
+        public void OnStop(string providerName, string activityName, int task, ref Guid activityId, bool useTplSource = true)
         {
             if (m_current == null)        // We are not enabled
                 return;
 
             string fullActivityName = NormalizeActivityName(providerName, activityName, task);
 
-            TplEventSource log = TplEventSource.Log;
-            if (log.Debug)
+            TplEventSource? log = useTplSource ? TplEventSource.Log : null;
+            bool tplDebug = log != null && log.Debug;
```

Comment:
```suggestion
            bool tplDebug = log?.Debug;
```

Reply:
I'd have to make it a nullable bool? (which I'd rather not do)